docs/content/guides/dagster/enriching-with-software-defined-assets.mdx

```diff
@@ -0,0 +1,407 @@
+| Every day, drop and recreate the `users` table and the `user_recommender_model` model that depends on it | Yes       | Assets are known before a run and are being updated              |
+| Every hour, add a partition to the `events` table                                                        | Yes       | Assets are known before a run and are being updated              |
+| Clicking a button refreshes the `recommender` model                                                      | Yes       | Assets are known before a run and are being updated              |
+| Every day, send emails to a set of users                                                                 | No        | No assets are being updated                                      |
+| Every day, read a file of user IDs and change the value of a particular attribute for each user          | No        | The set of assets to update is not known before running the job. |
+| Every day, scan my warehouse for tables that haven't been used in months and delete them                 | No        | The set of assets to update is not known before running the job. |
+
+---
+
+## How do software-defined assets work with other Dagster concepts?
+
+Not sure how software-defined assets fit into your current Dagster usage? In this section, we'll touch on how software-defined assets work with some of Dagster's core concepts:
+
+- [Ops and graphs](#ops-and-graphs)
+- [Jobs](#jobs)
+- [Dagster types](#dagster-types)
+- [Repositories](#repositories)
+- [IO managers](#io-managers)
+
+### Ops and graphs
+
+| Without software-defined assets                                           | With software-defined assets                                                                   |
+| ------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------- |
+| An [op]() is the basic unit of computation                                | Every software-defined asset includes a graph or an op                                         |
+| A [graph]() is a composite unit of computation that connects multiple ops | Every software-defined asset includes a graph or an op                                         |
+| Ops can have multiple outputs                                             | Multiple assets can be produced by a single op when defined using the `@multi_asset` decorator |
+| Ops can use [config](/concepts/ops-jobs-graphs/ops#op-configuration)      | Assets can use [config]()                                                                      |
+| Ops can be tested by directly invoking them                               | Assets can be tested by directly invoking them                                                 |
+| Ops can require resources                                                 | Software-defined assets can require resources                                                  |
```

Comment:
I think it would be helpful here to mention that ops & software defined assets both have access to `OpExecutionContext`